dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -182,6 +199,17 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                         THROW_IF_FAILED(notificationPlatform->AddToastRegistrationMapping(processName.get(), toastAppId.c_str()));
                     }
 
+                    // Need to register sink on new channel creation if handlers are registered
+                    bool registeredEvent{ false };
```

Comment:
I'm having a hard time understanding why this is here as opposed to REgister() call. I mean if a prior channel already exists, shouldnt the foreground handler stuff just work after Register is called?

This puts Channel request as a forcing function that seems incorrect.

Reply:
If a Channel is requested, the old sink is torn down. So what here is an example of what could happen.
1. App requests channel 
2. App closes and is launched by raw notification
3. Register foreground handlers and handle first notification in background
4. Request channel (old sink is now dead)
5. Second notification goes to background because foreground activation failed

Reply:
So 
Main()
1) Register event Handler
2) Manager::Register() (At this point, the event handlers should be active if we have a prior channel along with the Push Kind extension)
3) Channel request succeeds and we get a new channel which replaces old one
4) All new channel payloads are still routed to event handler

Reply:
Essentially every time a Channel is requested, a new foreground sink has to be registered.

Reply:
Step 3 removes the old notification sink and since PushReceived is never called again, there is no longer a foreground sink

Reply:
I need to think about this a bit more. The issue for me is what happens with a valid channel (from prior launch) and we get a new notification.

Reply:
Or lets say, we get 2 new notifications

Reply:
Can you explain what happens for packaged and unpackaged scenarios when channel request is not done, but REgister is called and we have an active channel?

Reply:
Everything is fine since the old sink still exists. But if the 2nd launch requests a channel, the old sink is torn down and the 3rd notification is sent to background.

Reply:
Valid channel (both scenarios):
1. PushNotification launches the app
2. Register FG handlers 
3. Register() and receive the first notification via background
4. Any notification after this is sent to foreground

5. Request Channel (old foreground sink is torn down)
6. All notifications will route to background because no foreground sink exists